src/slider/slider.js

```diff
@@ -0,0 +1,225 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+import * as React from 'react';
+import {findDOMNode} from 'react-dom';
+import {STATE_CHANGE_TYPE} from './constants';
+import type {ChangeActionT, StatelessStateT, PropsT} from './types';
+import {
+  Root as StyledRoot,
+  Axis as StyledAxis,
+  AxisRange as StyledAxisRange,
+  Tick as StyledTick,
+  TickBar as StyledTickBar,
+  Thumb as StyledThumb,
+} from './styled-components';
+import {getOverrides} from '../helpers/overrides';
+/* global document */
+/* global window */
+
+class Slider extends React.Component<PropsT, StatelessStateT> {
+  domNode: ?Element | Text;
+  onMouseUpListener: ?MouseEventHandler;
+  onMouseMoveListener: ?MouseEventHandler;
+  static defaultProps = {
+    overrides: {},
+    onChange: () => {},
+    onBlur: () => {},
+    onFocus: () => {},
+    onMouseEnter: () => {},
+    onMouseLeave: () => {},
+    onMouseDown: () => {},
+    onMouseUp: () => {},
+    error: false,
+    autoFocus: false,
+    tabIndex: 0,
+  };
+
+  state = {
+    isThumbMoving: false,
+    currentThumb: -1,
+    currentMove: 0,
+    thumbRefs: this.props.value.map(() => React.createRef()),
+  };
+
+  constructor(props: PropsT) {
+    super(props);
+  }
+
+  onFocus = (e: SyntheticEvent<HTMLInputElement>) => {};
+
+  onBlur = (e: SyntheticEvent<HTMLInputElement>) => {};
+
+  onMouseEnter = (e: SyntheticEvent<HTMLInputElement>) => {};
+
+  onMouseLeave = (e: SyntheticEvent<HTMLInputElement>) => {};
```

Comment:
what's the purpose of these noops above?

Reply:
removed